nelc/openedx-completion-aggregator
# Add tracking logs

**Description:** 

This PR updates https://github.com/open-craft/openedx-completion-aggregator/pull/173 to merge latest `master` and get tests passing again.

https://github.com/open-craft/openedx-completion-aggregator/pull/173 is the first step for https://github.com/openedx/openedx-aspects/issues/222

**JIRA:** 

Private-ref: [FAL-3766](https://tasks.opencraft.com/browse/FAL-3766)

**Merge deadline:** ASAP -- there's follow-up work to come soon.

**Installation instructions:**

1. Install this package in your LMS venv
2. Ensure completion is enabled : should be a waffle switch enabled for `[completion.enable_completion_tracking](http://local.edly.io:8000/admin/waffle/switch/1/change/)`
3. Create a course with one or two subsections, sections, blocks.

**Testing instructions:**

1. Watch the tracking.log file generated by your LMS for completion events, e.g `tail -f $(tutor config printroot)/data/lms/logs/tracking.log | grep compl`
4. Enrol in your test course and complete some blocks.
  You should see `openedx.completion_aggregator.progress.*` events coming through to mark the progress towards completion of the units (vertical), subsections (sequential), sections (chapter), and course.
  When you've completed the children of these blocks, you should see corresponding `openedx.completion_aggregator.completion.*` events.

**Reviewers:**
- [ ] @andrey-canon

**Merge checklist:**

**Post merge:**
      finished.
